sherpa/astro/xspec/src/_xspec.cc

```diff
@@ -133,262 +62,118 @@ static int _sherpa_init_xspec_library()
-    // to XSPEC stream buffers, that in turn talk to std::cin,
-    // std::cout, and std::cerr.  When we are not given a toggle
-    // to turn off output to output streams, we can instead redirect
-    // the stream itself.
-
-    // In this case, the only stream we want to redirect is stdout.
-
-    // If creating any stream buffer fails for any reason,
-    // make sure we can still use original std::cout stream buffer.
-
-    // Note:  since we are *not* redirecting std::cerr, any error
-    // messages are still seen; if, for example, HEADAS is not set,
-    // user will still see error message printed to stderr (i.e., to
-    // screen by default).  We still want such error messages to be
-    // seen!  So do *not* redirect std::cerr.
-    //
-    // Unfortunately it appears that XSPEC does not use stderr for its
-    // error messages, but stdout, so the following code will hide
-    // any error messages from FNINIT (XSPEC versions 12.8.2 and 12.9.0).
-    // Perhaps the screen output could be saved, rather than redirected
-    // to /dev/null, and checked for the string '***Error: ', which
-    // appears to be used in the error messages from XSPEC.
-
-    cout_sbuf = std::cout.rdbuf();
-    fout.open("/dev/null");
-    if (cout_sbuf != NULL && fout.is_open())
-      std::cout.rdbuf(fout.rdbuf()); // temporary redirect stdout to /dev/null
-
     // Initialize XSPEC model library
     FNINIT();
 
-    // Get back original std::cout
-    if (cout_sbuf != NULL) {
-      std::cout.clear();
-      std::cout.rdbuf(cout_sbuf);
-    }
-    fout.clear();
-    fout.close();
-
-    // We used to set the chatter to 0 but this misses useful info
-    // (like XSPEC can't find the data files) that would be reported
-    // by XSPEC, so use the default XSPEC setting. It does not appear
-    // that this value is read in from ~/.xspec/Xspec.init so set
-    // it here so we have repeatable behavior.
-    //
-    FPCHAT( 10 );
-
-    // Set cosmology initial values to XSPEC initial values
-    csmph0( 70.0 );
-    csmpq0( 0.0 );
-    csmpl0( 0.73 );
-
   } catch(...) {
-
-    // Get back original std::cout
-    if (cout_sbuf != NULL) {
-      std::cout.clear();
-      std::cout.rdbuf(cout_sbuf);
-    }
-    fout.clear();
-    fout.close();
+    IosHolder::setStreams(IosHolder::inHolder(),
+			  outStream,
+			  IosHolder::errHolder());
 
     // Raise appropriate error message that XSPEC initialization failed.
     PyErr_SetString( PyExc_ImportError,
 		     (char*)"XSPEC initialization failed; "
 		     "check HEADAS environment variable" );
     return EXIT_FAILURE;
-
   }
 
-  init = true;
-  return EXIT_SUCCESS;
-
-}
-
-static PyObject* get_version( PyObject *self )
-{
-
-  char version[256];
-  int retval;
-
-  try {
+  IosHolder::setStreams(IosHolder::inHolder(),
+			outStream,
+			IosHolder::errHolder());
 
-    retval = xs_getVersion(version, 256);
+  // We used to set the chatter to 0 but this misses useful info
```

Comment:
I would suggest changing 'misses' to something along the lines of drops or obscures since the 0 level chatter actually filters out the info as intended.